### PR TITLE
More porting

### PR DESCRIPTION
**Pull request by madhanrm**

### CODE REVIEW

**Review by raghavendra-nataraj — APPROVED** [review]

File: pkg/cluster/compute/resource/resourcecollection.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+func (instances *ResourceCollection) Close() (err error) {
+	for _, value := range *instances {
+		value.Close()
+	}
+	return nil
+}
+
+func (instances *ResourceCollection) String() string {
+	return ""
```
Comment (apluchik) [inline pkg/cluster/compute/resource/resourcecollection.go:32]: would we want to implement this in the future?

File: pkg/cluster/compute/internal/embeddedxml.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+func GetVMID(xmlContent string) (vmID string, err error) {
+	var clusterResource Instance
+	err = xml.Unmarshal([]byte(xmlContent), &clusterResource)
+	if err != nil {
+		return "", err
+	}
+
+	for _, propertyObject := range clusterResource.PropertiesObject {
+		if propertyObject.Name == "PrivateProperties" {
+			for _, property := range propertyObject.Value.Instance.Properties {
```
Comment (apluchik) [inline pkg/cluster/compute/internal/embeddedxml.go:65]: can these be nil?

**Review by apluchik — APPROVED** [review]

File: pkg/cluster/compute/internal/embeddedxml.go (hunk 2 above)
Comment (madhanrm) [reply pkg/cluster/compute/internal/embeddedxml.go:65]: I moved these code from moc-pkg. Havent changed anything.

File: pkg/cluster/compute/resource/resourcecollection.go (hunk 1 above)
Comment (madhanrm) [reply pkg/cluster/compute/resource/resourcecollection.go:32]: We dont use these. Been there for debugging. Will have to fix this across for all resources.

**Pull request merged**